components/amazon/amazon.go

```diff
@@ -17,54 +17,52 @@ type CreateAmazonMaster struct {
 
 type AmazonNodePool struct {
 	InstanceType string `json:"instanceType"`
-	SpotPrice string 		`json:"spotPrice"`
-	MinCount  int  		  `json:"minCount"`
-	MaxCount  int 	    `json:"maxCount"`
-	Image     string 		`json:"image"`
+	SpotPrice    string `json:"spotPrice"`
+	MinCount     int    `json:"minCount"`
+	MaxCount     int    `json:"maxCount"`
+	Image        string `json:"image"`
 }
 
 type UpdateClusterAmazon struct {
-	NodePools []*UpdateAmazonNodePool `json:"nodePools,omitempty"`
+	NodePools map[string]*UpdateAmazonNodePool `json:"nodePools,omitempty"`
 }
 
 type UpdateAmazonNodePool struct {
-	Name string		`json:"name"`
-	MinCount int 	`json:"minCount"`
-	MaxCount int 	`json:"maxCount"`
+	Name     string `json:"name"`
```

Comment:
The `Name` field is redundant now so I think it can be removed otherwise the json will look like:

`"nodePools": {
      "pool1" : {
          "name": "pool1",
          "minCount": 1,
          "maxCount": 2
      }
}`